xamarin/Xamarin.Forms
# Add Essentials dependency

### Description of Change ###

Adding the dependency on Essentials, adding the first call to the RequestedTheme API and a gallery page for that.

We have decided to take on the dependency on Essentials. We are seeing more and more areas where we would have potential overlap. In order not to duplicate functionality, we simply decided to add the dependency to Essentials and make use of their awesome stuff.

### Issues Resolved ### 


- related to #9804

### API Changes ###
 
 None

### Platforms Affected ### 
Not directly affected, but the dependency is added for these platforms to begin with:

- Core/XAML (all platforms)
- iOS
- Android
- UWP
- MacOS
- Tizen

### Behavioral/Visual Changes ###


None

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###
Take the NuGets resulting for this build, add them to a File > New project and/or existing project and see if everything proceeds to work as it did before.

### PR Checklist ###
